src/Microsoft.Data.Analysis/StringDataFrameColumn.BinaryOperations.cs

```diff
@@ -103,6 +103,11 @@ public PrimitiveDataFrameColumn<bool> ElementwiseEquals(string value)
 
         public override PrimitiveDataFrameColumn<bool> ElementwiseEquals<T>(T value)
         {
+            DataFrameColumn column = value as DataFrameColumn;
+            if (column != null)
+            {
+                return ElementwiseEquals(column);
```

Comment:
Are we certain this will bind to the `DataFrameColumn` overload? and it won't turn into infinite recursion? Make sure we have tests for all these to ensure passing in a DataFrameColumn works here.

Reply:
Yup, first thing I verified. The compiler is able to find the `DataFrameColumn` overload. It goes back to the generic overload only when a derived type is passed in, for ex: `StringDataFrameColumn`. Added some tests for this too.